lib/Lower/Bridge.cpp

```diff
@@ -373,15 +373,22 @@ class FirConverter : public Fortran::lower::AbstractConverter {
       std::string &name, const Fortran::semantics::Symbol *&symbol) {
     setCurrentPosition(stmt.source);
     auto &n = stmt.statement.v;
-    name = n.ToString();
     symbol = n.symbol;
+    assert(symbol && "Name resolution failure");
+    name = mangleName(*symbol);
   }
   void
   genFIR(const Fortran::parser::Statement<Fortran::parser::EndMpSubprogramStmt>
              &stmt,
-         std::string &, const Fortran::semantics::Symbol *&) {
-    genFIRProcedureExit(
-        static_cast<const Fortran::parser::MpSubprogramStmt *>(nullptr));
+         std::string &, const Fortran::semantics::Symbol *&symbol) {
+    setCurrentPosition(stmt.source);
+    assert(symbol);
+    if (symbol->get<Fortran::semantics::SubprogramDetails>().isFunction()) {
```

Comment:
Looking at this, I think it would be nice to have a set of "detail getters" in the bridge that get the (right) details from a symbol and encapsulate these `Fortran::semantics` details.

Reply:
I agree, but actually these getters should be in semantics tools. My bad here, there is already a `IsFunction(symbol)` function there. I've changed that to use it. Now, we still access details sometimes, to get things like dummy, shapes... I think it is OK, I am afraid that creating structures to hide the details will make our maintenance life harder in the bridge, we will duplicate utilities. It will make the life of people who works at the edge of semantics/bridges easier to only have to know one interface.

Reply:
Ah. I guess I did say "in the bridge", but I don't have a strong opinion on that. It would reduce typing to have something like
```
bool isFunction(SymbolRef symbol) { return Fortran::semantics::utility::IsFunction(symbol); }
```
but I agree that using the code in semantics (if it exists) is the best approach.